### ISSUE TITLE
6.0.0 broke pushes

### ISSUE THREAD

**Issue by CodySwannGT** [aws-amplify/amplify-cli#8221]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v14.17.5

### Amplify CLI Version

6.0.0

### What operating system are you using?

Mac

### Amplify Categories

auth

### Amplify Commands

update

### Describe the bug

Trying to add oAuth integration via amplify update auth broke our project

### Expected behavior

We're able to add oAuth without breaking our project

### Reproduction steps

1. ``$ amplify update auth``

produces: 

````
TypeError: Cannot read property 'length' of undefined
    at Object.run (/Users/cody/.nvm/versions/node/v14.17.5/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-auth/lib/commands/auth/update.js:38:56)
    at Object.executeAmplifyCommand (/Users/cody/.nvm/versions/node/v14.17.5/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-auth/lib/index.js:294:25)
    at /Users/cody/.nvm/versions/node/v14.17.5/lib/node_modules/@aws-amplify/cli/lib/execution-manager.js:180:51
    at executePluginModuleCommand (/Users/cody/.nvm/versions/node/v14.17.5/lib/node_modules/@aws-amplify/cli/lib/execution-manager.js:164:11)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Object.executeCommand (/Users/cody/.nvm/versions/node/v14.17.5/lib/node_modules/@aws-amplify/cli/lib/execution-manager.js:35:9)
    at async Object.run (/Users/cody/.nvm/versions/node/v14.17.5/lib/node_modules/@aws-amplify/cli/lib/index.js:162:9)
````

2. To workaround, we manually added ``"aliasAttributes": "",`` to parameters.json
3. ``$ amplify push`` 

produces: 

````Existing schema attributes cannot be modified or deleted.````

4. Removed ``"aliasAttributes": "",`` from parameters.json
5. ``$ amplify push``

produces:

```Parameters: [aliasAttributes] must have values````

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by josefaidt** [comment]
Hey @CodySwannGT what version of the CLI was used to create the auth resource? [Version 6.0.0 includes changes to the auth resource creation](https://docs.amplify.aws/cli/migration/cli-auth-signup-changes/)

**Comment by CodySwannGT** [comment]
Thanks for the quick reply. We got further with that, but still no dice.

We added the feature flag, then ran amplify update auth

But when we pushed, we got: 

````Existing schema attributes cannot be modified or deleted. ````

[Here's a full video](https://www.dropbox.com/s/3nys8hyq39y0xfx/amplify-1.mp4?dl=0)

**Comment by Straubulous** [comment]
Hi CodySwannGT - sorry to hear about the issue.  With the feature flag set, can you try putting the "aliasAttributes": ""; back into parmeters.json and then push again?

**Comment by josefaidt** [comment]
Hey @CodySwannGT thanks for sending that video over! We're actively investigating. To clarify, what version of the CLI was used to initially create the auth resource?

**Comment by CodySwannGT** [comment]
@josefaidt - Thanks really appreciate it because we're stuck!

This app was originally created in June 2019. So the auth was definitely originally created with version <= 5.1.2.

We try to stay current on releases, but we definitely haven't made any updates to the auth category in at least a year.

The auth category is a different story, and we made changes to that just the other day (with v6.0.0). Those deployed just fine.

**Comment by josefaidt** [comment]
Thanks for the swift response @CodySwannGT. It appears there are three issues occurring here:

1. The `HostedUI...` issue on push after updating auth with social providers
2. when running `amplify update auth`, we ran into this issue where the CLI was checking the parameters for `aliasAttributes`. We were able to workaround this by adding `"aliasAttributes": []` in the auth resource's `parameters.json` file
3. running `amplify auth update` is unintentionally removing custom attributes from the auth resource's `parameters.json` file and/or the CloudFormation template

I noticed the removed custom attributes from your git diff screenshot. Can you confirm whether the CloudFormation template was also modified after the `amplify auth update`?

**Comment by CodySwannGT** [comment]
You're correct on the the third point. 

I've tried going through this a number of times.

The first time, after running ``amplify auth update``, I **DID** add back the custom attributes that were stripped out.

The second time, after running ``amplify auth update``, I **DID NOT** add back the custom attributes that were stripped out.

In both circumstances, I got the ``Existing schema attributes cannot be modified or deleted.`` error.

**Comment by josefaidt** [comment]
When you added the custom attributes back were there any modifications to the CloudFormation template that you also had to reverse? I'm currently running through the reproduction

**Comment by CodySwannGT** [comment]
No. No other modifications. I simply put back the custom attributes (which, side note, that has been a known issue we've just dealt with for as long as I can remember. Annoying, but not a blocker)

**Comment by CodySwannGT** [comment]
UPDATE: I think we might be on to something.

In the cloudformation-template.yml file, under the UserPool block, the Schema block was changed by the ``amplify update auth`` to remove the custom attributes.

````
- Name: zip
          Required: false
          Mutable: true
          AttributeDataType: String

        - Name: address2
          Required: false
          Mutable: true
          AttributeDataType: String

        - Name: city
          Required: false
          Mutable: true
          AttributeDataType: String

        - Name: state
          Required: false
          Mutable: true
          AttributeDataType: String

        - Name: mobile
          Required: false
          Mutable: true
          AttributeDataType: String

        - Name: referral_code
          Required: false
          Mutable: true
          AttributeDataType: String
        
        - Name: bonus_code
          Required: false
          Mutable: true
          AttributeDataType: String
````

We put those back but are now getting this error:

`` Role does not have permission to publish with SNS``

Looking into that now.

**Comment by CodySwannGT** [comment]
OK - the last part was a problem with our Boundary permission, so we fixed that and I think we're cooking with gas.

Fix steps:

1. Downgrade to 5.1.2 - ``npm install -g @aws-amplify/cli@5.1.2``
2. Update auth category to add oAuth - ``$ amplify update auth``
3. Add custom parameters back into parameters.json AND xxxxxx-cloudformation-template.yml
4. Deploy - ``$ amplify push``
5. Add ``"forceAliasAttributes": true`` to cli.json
6. Upgrade to 6.0.0 - ``npm install -g @aws-amplify/cli``

Admittedly, part of this was an oversight on our end, not updating custom attributes in both files.

My ask would be:

1. Add a link in the changelog that explains breaking changes (as 6.0.0 had)
2. Add a way to manage custom user pool attributes from amplify directly

Thank you so much for your help on this - you may close if you'd like!

**Comment by josefaidt** [comment]
Ah thanks for the update! It does appear that adding/maintaining custom attributes should have a better experience as outline in this existing feature request: https://github.com/aws-amplify/amplify-cli/issues/1204. That issue also has the pain points experienced when updating auth resources that have these values present. Despite the initial issue where the CLI was looking for `aliasAttributes` upon running `amplify update auth`, I'm glad you were able to find a workaround. 

I'll keep this open to track the original issue as it has been identified as a bug with this latest release. 🙂

**Comment by Straubulous** [comment]
A fix for this issue has been published in Amplify latests (6.0.1). Closing.

**Comment by chrisl777** [comment]
After running `amplify update auth` (CLI version 6.0.1), for some reason the CLI deleted this block from my Auth-cloudformation-template.yml: 

```
AliasAttributes:
        - email
        - phone_number
```

I added this block back and did an `amplify push` and I no longer received an error about updated AliasAttributes.

Note that, according to the Cognito docs, AliasAttributes cannot be modified. 

<img width="829" alt="Screen Shot 2021-09-24 at 11 36 20 PM" src="https://user-images.githubusercontent.com/230479/134756528-6ed97494-20d2-4c58-8619-b4a349e66b6f.png">
